pkg/server/v2/gateway.go

```diff
@@ -57,27 +58,24 @@ func (h *HTTPGateway) ServeHTTP(resp http.ResponseWriter, req *http.Request) err
 	case resource.FetchRuntimes:
 		typeURL = resource.RuntimeType
 	default:
-		http.Error(resp, "no endpoint", http.StatusNotFound)
-		return fmt.Errorf("no endpoint")
+		return nil, http.StatusNotFound, fmt.Errorf("no endpoint")
 	}
 
 	if req.Body == nil {
-		http.Error(resp, "empty body", http.StatusBadRequest)
-		return nil
+		return nil, http.StatusBadRequest, fmt.Errorf("empty body")
 	}
 
 	body, err := ioutil.ReadAll(req.Body)
 	if err != nil {
-		http.Error(resp, "cannot read body", http.StatusBadRequest)
-		return nil
+		return nil, http.StatusBadRequest, fmt.Errorf("cannot read body")
 	}
 
 	// parse as JSON
 	out := &discovery.DiscoveryRequest{}
-	err = jsonpb.UnmarshalString(string(body), out)
+	unmarshaller := jsonpb.Unmarshaler{AllowUnknownFields: true}
```

Comment:
@kyessenov Do you think its acceptable to go with `jsonpb.Unmarshaler{AllowUnknownFields: true}` to get around the issue of unknown fields? I've created an issue in envoy https://github.com/envoyproxy/envoy/issues/10633, and we can remove this special case when the fix lands.